README.md
# Themia

针对typecho优化的Themia模板（Rhilip个人修改版本）

下载后上传到typecho模板目录解压，将解压后的文件夹重命名为Themia，即可使用

模板特色：

无需插件的文章置顶功能，
无需插件显示文章浏览次数(有cookie验证)，
无需插件大赏跨屏浏览功能，
无需插件的缩略图功能，
版权保护功能，界面语言中/英切换，文章内容简/繁翻译，
模板样式的自定义功能，超强的自定义字段功能，利用QQ空间，新浪等api的分享功能


CHANGELOG.md
# 更新日志

> 2018.8.8 v3.8.1

 - 初次提交本人日常使用过程中对模板的修改。
 - 增加富强插件，增加版权说明。（前端展示）
 - 优化`404.php`页面js。
 - 修改原`新浪微博`为 `Github`。

# 更新内容缺失 （3.7.6 至 3.8.0）

# 原作者更新内容（至3.7.6）

> 2017-09-21 3.7.6 

 - 文章二维码api更换，原生评论样式更换解决评论bug，部分提示文字修整。

> 2016-09-13 3.7.4 
 
 - 站点成立时间统计代码优化，兼容更多的浏览器，同时显示位置改为站点底部显示。
 
> 2016-08-10 3.7.3 

 - fm 字段维护，css 轻改，调整文章内容与封面的边距，同时 #header 部分加入半透明效果。
 - author 页面内加入用户组显示。

> 2016-07-16 3.7.2 

 - 修复首行缩进代码导致的评论等布局错乱问题。
 - 文章阅读次数显示方案非插件实现方式中加入了 cookie 验证，在不清除 cookie 或者 cookie 未过期的情况下不会重复计数，实现方法参考了《 绛木子 TePostViews 插件》
 - 修复由于前期版本代码整理导致的独立页面编辑按钮 404 问题。
 - 由于部分网友主机对于 isset() 不太支持，故删掉此方法，这个变动会引起一些设置问题，以前自定义字段的字段值可以不填的，现在需要随便填些东西占位。
 - 修复 li 字段，样式未加载问题。、
 - 居中风格优化。
 - 评论列表中头像加入爬墙 (需要后台开启，如开启后出现 bug 的话，请关闭此项)

> 2016-06-29 3.7.1

 - author.php 页面的文章列表数目改为 50，也就是说超过 50 篇文章就会分页。
 - author.php 的 gravatar 头像，加入自动爬墙。
 - 评论样式调整，头像加入爬墙。
 - 分享功能人人网分享改为谷歌分享 (上次更新的有遗漏地方)。
 - 字段 st 剔除，修复独立页面自定义字段 bug。
 - 后台网站主体风格增加一种‘居中风格’，可配合自定义字段 m 使用，也可选择强制所有文章使用 more 语法截取缩略内容。
 - 加入文章目录自动生成功能，需要在模板后台开启。

> 2016-06-24 3.7.0
 - 打赏与文章二维码样式调整
 - 分类归档对父级 / 子级分类布局支持，同时定位方式改用缩略名。
 - 精简代码 (修改了好多 php 文件，最浪费时间啦)
 - 修复部分主机缩略图不显示问题 (对空格比较敏感的主机)。
 - 增加 author.php 页面
 - 加入文章加载提示文字
 - 搜索功能调整
 - 分享功能人人网分享改为谷歌分享
 - 修复部分情况下页面可横向滚动问题
 - 增加 1080P 等大屏显示器下的 SideBar 样式
 - 调整文章的样式，比如字体、字体大小、对齐等
 - 中英混排自动加入空格（参考 pangu.js）
 - 等等........

> 2016-06-05 3.6.5

 - 加入一个自定义字段：实现Li标签大屏幕下多列显示，效果见我的友情链接页。
 - 相册功能加入更优雅的实现方式。
 - 修复chrome浏览器的站标显示的小问题。
 - 图片灯箱效果取消，默认是不加入灯箱效果的，如果启用灯箱效果需要在编辑器图片后面加入"+"号。(这样设置的好处就是你用和图片有关的插件时不会冲突)
 - 使用自定义字段'm'可使用more语法截取缩略文章，不懂more的，请忽略该功能。
 - archives.php时间归档不容易被触发的奇葩bug修复。
 - 细节优化,取消发呆....。
 - 文章二维码js生成改为百度API生成。

> 2016-05-18 3.6.3

 - css调整：除首页面外的页面超链接带下划线了。
 - 缩略图功能优化，解决某些编辑器图片提取不出来的问题。
 - 时间归档页，搜索功能小bug修复。
 - 模板搜索功能加入搜索结果为空提示。

> 2016-05-07 3.6.2

 - 图片灯箱效果增加alt属性，对于图片性文章，对于seo有利。
 - css修复个问题。
 - index.php文件代码轻调整。
 - 模板js文件调整，体验更丝滑，同时修复ie11和火狐下简繁翻译的一个bug。
 - 文章内容中的超链接自动加target="_blank"。
 - 无关紧要的东西：模板后台预览图改为我亲手ps的图片，离开当前标签显示的ico图标也有所改动。
 - 首行缩进功能，模板后台开启后，文章内容首行将自动缩进2em(两个汉字距离)。
 - 解决与ShortLinks插件冲突问题。

> 3.6.1 略

> 2016-04-18 3.6

 - 一些没卵用的功能进行删减：日文界面选项，jiathis，bshare的API文章生成二维码功能删掉。
 - QQ空间,新浪微博，人人网分享功能改为引用各自的分享Api，分享后可自带文章的缩略图。
 - 缩略图提取顺序纠正：附件第一张图->文章第一张图->随机一张图。
 - 无分类无标签的文章标签显示怪异修复,有多个分类没有标签时标签处怪异修复。
 - 【编辑器内打出“{你 好|ni hao}”←即可实现拼音注释功能】
 - 字体方面优化，部分中文界面宋体改为Helvetica,如果不存在及默认微软雅黑，如果微软雅黑不存在就宋体。
 - 加入了本地评论，样式可能不太友好，如果不需要的话可以删除comments.php里面的内容，然后填入第三方评论代码

> 2016-04-09 3.5.1

 - 当typecho文章没有标签时文章标签调用分类。
 - css将图标css独立出来方便大家自行升级图标。
 - 修复文章分类空白：【（触发这个问题，需要你增加个文章，为文章选择个分类，发布文章，最后去后台把这个分类删除）或者（分类里面没有设置默认分类，增加文章后，没有选择任何分类，然后发布文章）】。你就会发现文章的分类消失了，显示为空，这次修复，会让这个空变为提示文字，来提醒博主这个文章的分类被删除了，文章无家可归了。
 - 优化字体加载不全问题。
 - 删除多余代码，优化js加载规则。

> 2016-03-28 3.2.0

 - 跨屏浏览二维码生成功能增加jiathis和bshare生成方式，优点是模板无需加载js等，缺点是依赖第三方api，也不能算是缺点，本模板的分享功能就是依赖的jiathis，也可以选择引用js生成的方式，默认是引用js生成，优点是稳定，缺点是模板需要多加载个js。(jiathis和bshare的api生成二维码速度也很快，直观上感觉不到差异。)
 - 文章中返回顶部按钮，增加过度效果。
 - 删除自带的代码高亮功能（久经思考还是决定删除了！还是建议用这个插件Prismjs插件有什么问题可以直接与作者反馈，这样方便插件作者更新！因为代码高亮输入格式都是一样的！所以安装了插件也不用去挨个修改旧的文章）。

> 2016-03-20 3.1.0

 - 修正模板某个自定义的样式与分享组件矛盾导致的显示错乱。(这个是最初弄模板时留下的问题，才发现问题所在，汗死)。
 - 分享组件，国内化，改为微博分享，人人网分享，QQ空间分享。
 - 修复图片与打赏组件直接接触会导致打赏插件会错乱。
 - blockquote样式优化，感谢KohinataMiku建议。

> 2016-03-14

 - 给文章中图片加入灯箱效果。
 - 界面语言设置，支持中文，英文，日本语（默认英文）。
 - 加入代码高亮功能，代码高亮功能改自Prismjs插件。
 - 修正某个严重错误(这个错误可能导致模板启动失败，大概)。
 - 文章页面内，前一篇文章后一篇文章按钮微调整。
 - 侧边菜单图标微调，更加舒适。
 - 分类归档时间显示问题修复！感谢KohinataMiku提醒。
 - 随机缩略图增加到99张，以后不再加了，保持这个数量。

> 2016-03-09 2.5.0

 - 文章置顶功能(模板后台，添加需要置顶文章的id即可)。
 - 后台错别字修正。
 - 优化博客加载速度。
 - 修正简繁切换问题。
 
> 2016-03-05 2.4.0版更新与改变内容

 - 打赏(支付宝二维码)，跨屏浏览功能（文章显示二维码），文章版权声明。
 - 增加文章自定义字段，实现自定义文章缩略内容(否则则是默认的截取文章前140个字符，可自行更改index.php，关于--more--语法，果然感觉还是不好，所以pass)。
 - 增加文章自定义字段，使文章缩略图显示位置显示在文章内容上方(否则默认是显示在文章内容右侧)。
 - 修复前几个版本footer.php 里面的Typecho的拼写错误(汗死！感谢哈陆lu提醒)。
 - sidebar.php博客名称调用的是用户名，改为调用博客名称(思考了下，发现我以前的想法有问题，才会调用用户名称)。
 - 博客背景，可在博客模板后台设置成纯颜色或者图片(不选择的话默认还是图片)。
 - 分类归档,超链接优化(非常小的改变，不过为了凑字数还是写出来了ヾ(o◕∀◕)ﾉ)。
 - 默认头像换为4号萌妹子，因为是2.4.0版嘛，(而且原来的默认头像太占地方了，170k为了一个头像貌似不值)。

> 2016-03-01 2.3.0更新内容

 - 文章阅读次数，可选择非插件支持，或者选择绛木子TePostViews插件(非插件的方案和Hanny的Stat插件使用的是同一个数据，所以如果你曾经用的是Star插件，可以直接选择第一项，同时禁用Star插件，以免重复计数【不禁用的话，访问一次，计数器会每次加2】)。
 - 分类归档优化，点击分类会跳到该页面下的对应文章处，评论数显示改为时间显示（因为评论数计算的是本地评论，然而模板用的是第三方评论）。
 - 时间归档，归档页面加入检索时间功能，输入时间，检索该页对应时间的文章，排版轻度改变。
 - about页优化，overflow:scroll 改为 overflow:auto，感谢KohinataMiku建议。
 - 版权声明处代码优化，且模板后台加入启动开关，默认是不启动的。
 - 人性化优化，文章编辑按钮点击后在新窗口打开，文章自带随机缩略图优化压缩内存，增加图片。
 - 文章标签显示处，一个标签一个按钮样式的显示方式(以前是不会弄，现在灵机一动，发现很简单就可以搞定)，还有就是删除了独立页面标签（独立页面根本不能添加标签，也不知道我为啥给加个显示标签的东西）。

404.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>404 Page Not Found</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style>
        * {
            margin: 0;
            line-height: 1.5;
        }

        html {
            color: #888;
            text-align: center;
            font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
        }

        body {
            position: absolute;
            top: 50%;
            left: 50%;
            margin: -43px 0 0 -150px;
            width: 300px;
        }

        h1 {
            color: #555;
            font-weight: 400;
            font-size: 2em;
        }

        p {
            line-height: 1.2;
        }

        @media only screen and (max-width: 270px) {
            body {
                position: static;
                margin: 10px auto;
                width: 95%;
            }

            h1 {
                font-size: 1.5em;
            }
        }
    </style>
</head>
<body>
<h1>404 Page Not Found</h1>
<p>页面不存在或被管理员删除， <span id="totalSecond">5</span>秒后自动返回首页</p>
</body>
<script language="javascript" type="text/javascript">
    let is_ie = navigator.appName.indexOf("Explorer") > -1;
    let second = document.getElementById('totalSecond').textContent;
    if (is_ie) {  //判断是IE浏览器还是Firefox浏览器，采用相应措施取得秒数
        second = document.getElementById('totalSecond').innerText;
    }
    setInterval("redirect()", 1000);  //每1秒钟调用redirect()方法一次
    function redirect() {
        if (second < 0) {
            location.href = '<?php $this->options->siteUrl(); ?>';
        } else {
            second -= 1;
            if (is_ie) {
                document.getElementById('totalSecond').innerText = second;
            } else {
                document.getElementById('totalSecond').textContent = second;
            }
        }
    }
</script>
</html>

sidebar.php
 <?php if($this->is('post') || $this->is('page')): ?> 


 <nav id="sidebar" <?php if ($this->fields->fm){ ?>data-behavior="3"<?php }else{ ?>
 <?php if ($this->fields->ys){ ?>data-behavior="<?php $this->fields->ys(); ?>"<?php }else{ ?>

data-behavior="<?php $this->options->css(); ?>"<?php };?><?php };?>>

<?php else: ?>
 <nav id="sidebar" data-behavior="<?php $this->options->css(); ?>" >
<?php endif; ?> 
 <div class="sidebar-profile">
            <a href="/#about">
                    <img class="sidebar-profile-picture" <?php if ($this->options->logoUrl){ ?>src="<?php $this->options->logoUrl();?>"<?php }else{ ?>src="<?php $this->options->themeUrl('image/avatar.jpg'); ?>"<?php };?>/> </a>
        <span class="sidebar-profile-name"><?php $this->options->title(); ?></span> 
        </div>
    
    
        <ul class="sidebar-buttons">
        
            <li class="sidebar-button">
                
                    <a  class="sidebar-button-link " href="<?php $this->options->siteUrl(); ?>" >
                
                    <i class="sidebar-button-icon fa fa-lg fa-home" ></i>
                    <span class="sidebar-button-desc"><?php if ($this->options->cdl == '0'): ?>Home<?php endif; ?><?php if ($this->options->cdl == '1'): ?>主页<?php endif; ?></span>
                </a>
        </li>
        
            <li class="sidebar-button">
                
                    <a  class="sidebar-button-link "
                         href="<?php $this->options->Categories();?>"
                        
                    >
                
                    <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
                    <span class="sidebar-button-desc"><?php if ($this->options->cdl == '0'): ?>Categories<?php endif; ?><?php if ($this->options->cdl == '1'): ?>分类归档<?php endif; ?>
</span>
                </a>
        </li>
        
            <li class="sidebar-button">
                
                    <a  class="sidebar-button-link "
                         href="<?php $this->options->Archive();?>"
                        
                    >
                
                   <i class="sidebar-button-icon fa fa-lg fa-calendar"></i>
                    <span class="sidebar-button-desc"><?php if ($this->options->cdl == '0'): ?>Archive<?php endif; ?><?php if ($this->options->cdl == '1'): ?>时间归档<?php endif; ?></span>
                </a>
        </li>
        
            <li class="sidebar-button"  id="sou">
                
                    <a  class="sidebar-button-link st-search-show-outputs" >
                
                    <i class="sidebar-button-icon fa fa-lg fa-search"></i>
                    <span class="sidebar-button-desc"><?php if ($this->options->cdl == '0'): ?>Search<?php endif; ?><?php if ($this->options->cdl == '1'): ?>搜索<?php endif; ?>
</span>
                </a>
        </li>
   
            <li class="sidebar-button">
                
                    <a  class="sidebar-button-link "
                         href="<?php $this->options->about();?>"
                        
                    >
                
                    <i class="sidebar-button-icon fa fa-lg fa-info-circle"></i>
                    <span class="sidebar-button-desc"><?php if ($this->options->cdl == '0'): ?>About<?php endif; ?><?php if ($this->options->cdl == '1'): ?>关于<?php endif; ?></span>
                </a>
        </li>
        
    </ul>
    
        <ul class="sidebar-buttons">
 
         
                   <li class="sidebar-button">
                
                    <a  class="sidebar-button-link " href="<?php $this->options->github();?>" target="_blank" >
                
                    <i class="sidebar-button-icon fa fa-lg fa-github"></i>
                    <span class="sidebar-button-desc"><?php if ($this->options->cdl == '0'): ?>Github<?php endif; ?><?php if ($this->options->cdl == '1'): ?>Github<?php endif; ?></span>
                </a>
        </li>

<li class="sidebar-button">
                
                    <a class="sidebar-button-link " href="<?php $this->options->links();?>">
                
                      <i class="sidebar-button-icon fa fa-lg fa-chain"></i>
                    <span class="sidebar-button-desc"><?php if ($this->options->cdl == '0'): ?>Links<?php endif; ?><?php if ($this->options->cdl == '1'): ?>友链<?php endif; ?></span>
                </a>
        </li>
    </ul>
    
</nav>


functions.php
<?php
if (!defined('__TYPECHO_ROOT_DIR__')) exit;
define("Themia_Version", "3.8.1");
function themeConfig($form) {
  echo '<p style="font-size:16px;text-align:center;">感谢您使用TE响应式主题 :<font color="#4A89DC"> Themia</font><font color="#F40"> '.Themia_Version.'</font> ![<a href="http://qqdie.com/archives/with-the-help-of-themia-subject-to-update-the-manual" target="_blank">帮助与更新</a>]</p>';
  //网站LOGO
    $logoUrl = new Typecho_Widget_Helper_Form_Element_Text('logoUrl', NULL, NULL, _t('博主头像地址'), _t('logo头像地址，不填写默认内置头像，这个建议用正方形的图片√'));
    $form->addInput($logoUrl);
  //背景
    $bgs = new Typecho_Widget_Helper_Form_Element_Radio('bgs',array('1' => _t('博客背景使用图片'),'2' => _t('博客背景使用纯色')),'1',_t('背景设置'),_t("选择这里的背景方案，对应填写下方的‘背景图or色’，这里默认是使用图片,设置图片好看，设置颜色提高网速，看个人情况设置吧"));
    $form->addInput($bgs); 

    $bgUrl = new Typecho_Widget_Helper_Form_Element_Text('bgUrl', NULL, NULL, _t('背景图or色'), _t('背景设置如果设置图片，这里就填写图片地址，如http://qqdie.com/1.jpg,背景设置如果选择的是颜色，这里就填写颜色代码如#000,这里如果不填写则默认图片或者颜色#444444'));
    $form->addInput($bgUrl);   

    $sticky = new Typecho_Widget_Helper_Form_Element_Text('sticky_1',NULL, NULL,'置顶文章ID', '填写对应主题的 id 即可使某些分类的文章在置顶首页显示（例如 1）'); //，多个文章以英文逗号分割（例如 1,2,3）');
    $sticky->input->setAttribute('class', 'mini');
    $form->addInput($sticky->addRule('isInteger', '请填入数字'));
 //文章首行缩进
    $page_suo = new Typecho_Widget_Helper_Form_Element_Radio('page_suo', 
    array(
        'display' => _t('post和page页面启用'),  
        'display1' => _t('仅post页面启用'),  
        'display2' => _t('仅page页面启用'),    
        'close' => _t('禁用')
        ), 
    'close', 
    _t('文章首行缩进'), _t('开启后，所有文章内容将自动缩进2em（两个中文汉字宽度）'));
    $form->addInput($page_suo);
    $github = new Typecho_Widget_Helper_Form_Element_Text('github', NULL,'https://github.com/Rhilip', _t('Github地址'), _t('填写你的Github主页地址到菜单目录中'));
    $form->addInput($github);

$Categories = new Typecho_Widget_Helper_Form_Element_Text('Categories', NULL,NULL, _t('Categories分类归档地址'), _t('新建独立页面，选择模板Categories，这里填入独立页面的完整地址'));
    $form->addInput($Categories);

$Archive = new Typecho_Widget_Helper_Form_Element_Text('Archive', NULL,NULL, _t('Archive时间归档地址'), _t('新建独立页面，选择模板Archive，这里填入独立页面的完整地址'));
    $form->addInput($Archive);

$links = new Typecho_Widget_Helper_Form_Element_Text('links', NULL,NULL, _t('links友情链接地址'), _t('新建独立页面，写上朋友的链接，作为友情链接页面，然后把页面地址填入这里'));
    $form->addInput($links);
$about = new Typecho_Widget_Helper_Form_Element_Text('about', NULL,NULL, _t('about链接地址'), _t('新建独立页面，写上关于自己的属性，然后把页面地址填入这里'));
    $form->addInput($about);
 $cdl = new Typecho_Widget_Helper_Form_Element_Radio('cdl',array('0' => _t('English'),'1' => _t('中文')),'0',_t('界面语言设置'),_t("尽管英文逼格比较高，但是中文有中文的好处。此处默认是英文。"));
    $form->addInput($cdl); 

$css = new Typecho_Widget_Helper_Form_Element_Radio('css',
array(
'1' => _t('1.在大屏幕显示宽的SideBar，在中屏幕显示窄的SideBar，在小屏幕显示抽屉SideBar（宽）'),
'2' => _t('2.在大屏幕和中屏幕显示窄的SideBar，在小屏幕显示抽屉SideBar（窄）'),
'3' => _t('3.在所有屏幕显示抽屉SideBar（宽）'),
'4' => _t('4.在所有屏幕显示抽屉SideBar（窄）'),
),
'1',
_t('sidebar风格'), _t('这里的风格指的是网站整体风格，文章风格可通过自定义字段字段名ys和字段值来控制，实现对单独的一篇文章的风格重定义（字段值就是上边对应的数字，如风格3，就在文章字段名填入ys，字段值填入3即可），文章封面效果也是用自定义字段，字段名fm，字段值就是封面的图片地址，启动封面效果的文章风格会变更为3（优先级高）√')); 
$form->addInput($css->multiMode());
//居中风格
  $tf = new Typecho_Widget_Helper_Form_Element_Radio('tf', 
    array(
        'open' => _t('启用居中风格，使用默认方式截取缩略内容'),  
        'more' => _t('启用居中风格，并强制使用more语法截取缩略内容'), 
        'close' => _t('不启用')
        ), 
    'close', 
    _t('文章标题居中风格'), _t('默认不启用，启用后文章缩略图会关闭显示，并且文章标题，分类等信息居中，该风格启用后，建议关闭置顶功能，否则会影响美观度。'));
    $form->addInput($tf);
    //工具开关
    $sidebarBlock = new Typecho_Widget_Helper_Form_Element_Checkbox('sidebarBlock', 
    array('kp' => _t('文章中加入跨屏浏览二维码，支付宝打赏，及文章版权声明')
,
'mulu' => _t('开启文章目录树功能'),
	'bjq' => _t('文章内容增强，加入最后编辑时间'),
	'Showfull' => _t('手机浏览器强制全屏，仅限于uc和QQ浏览器')
,'Showcolor' => _t('手机谷歌浏览器状态栏颜色渲染为白色'),
	'kiana' => _t('kiana封印解除'),
	'simg' => _t('全局不显示文章缩略图'),
	'pltx' => _t('评论列表头像自动翻墙')
,),
    array('Showcolor','bjq'), _t('工具开关'));
    $form->addInput($sidebarBlock->multiMode());


  $zfbUrl = new Typecho_Widget_Helper_Form_Element_Text('zfbUrl', NULL, NULL, _t('支付宝收款二维码'), _t('这里添加收款二维码的图片地址，不添加则默认jrotty的支付宝二维码'));
    $form->addInput($zfbUrl);

$wxUrl = new Typecho_Widget_Helper_Form_Element_Text('wxUrl', NULL, NULL, _t('微信收款二维码'), _t('这里添加收款二维码的图片地址，不添加则默认jrotty的微信二维码'));
    $form->addInput($wxUrl);

  $jsq = new Typecho_Widget_Helper_Form_Element_Radio('jsq',array('0' => _t('不显示文章浏览次数'),'1' => _t('非插件实现'),'2' => _t('绛木子TePostViews插件')),'1',_t('文章阅读次数显示方案（最后一项需要自行安装对应插件）'),_t("在工具开关中，打开文章浏览次数，然后选择这里的方案，两款方案最终效果都一样<br>只有绛木子TePostViews插件，在不清除cookie或者cookie未过期的情况下不会重复计数<br>提示：非插件的方案和Hanny的Stat插件使用的是同一个数据，所以如果你曾经用的是Star插件，可以直接选择非插件项，同时禁用Star插件，以免重复计数【不禁用的话，计数器计数会翻倍】"));
    $form->addInput($jsq); 

    $bqg = new Typecho_Widget_Helper_Form_Element_Radio('bqg',array('1' => _t('不开启版权狗'),'2' => _t('开启版权声明')),'1',_t('版权声明'),_t("对于未登录的用户复制文章内容超过42个字符时，复制的内容会自带版权声明。"));
    $form->addInput($bqg); 
 //建站时间
    $time = new Typecho_Widget_Helper_Form_Element_Text('time', NULL, '2015/06/06', _t('博客成立时间'), _t('在这里填入博客的成立时间,格式要求，完整如填入“2015/06/06 00:00:00”或者只填写年月日“2015/06/06”。'));
    $form->addInput($time);

   //统计代码
$tongji = new Typecho_Widget_Helper_Form_Element_Textarea('tongji', NULL,'统计代码', _t('备案信息+统计代码'), _t('填入备案信息和cnzz等第三方统计代码'));
$form->addInput($tongji);

}

/**
 * 解析内容以实现附件加速
 * @access public
 * @param string $content 文章正文
 * @param Widget_Abstract_Contents $obj
 */
function parseContent($obj){
    $options = Typecho_Widget::widget('Widget_Options');
    if(!empty($options->src_add) && !empty($options->cdn_add)){
        $obj->content = str_ireplace($options->src_add,$options->cdn_add,$obj->content);
    }
    echo trim($obj->content);
}
function themeInit($archive)
{
	 if ($archive->is('author')) {
 $archive->parameter->pageSize = 50; // 自定义条数
}
    if ($archive->is('single'))
    {
        $archive->content = image_class_replace($archive->content);
    }
}

function image_class_replace($content)
{
 $content = preg_replace('#<a(.*?) href="([^"]*/)?(([^"/]*)\.[^"]*)"(.*?)>#',
        '<a$1 href="$2$3"$5 target="_blank">', $content);



  $content = preg_replace('#<img(.*?)src="([^"]*/)?(([^"/]*)\.[^"]*)"([^>]*?)>\+#',
        '<div class="figure nocaption fig-100"><p><a class="fancybox" href="$2$3" title="" target="_blank" rel="external"><img$1 class="fig-img" src="$2$3" $5>
</a></p></div>', $content);



$content = preg_replace('#\[t(:|：)(.*?)\]<br\s*/?>#',
        '<div class="image-gallery">
        <div class="image-gallery-metabar">
            <span>$2</span>
        </div>', $content);

$content = preg_replace('#<img(.*?)src="([^"]*/)?(([^"/]*)\.[^"]*)"([^>]*?)>上(<br\s*/?>)?#',
'<div class="image-gallery-photos image-gallery-photos--thumbnail"> <div class="photo-box">
                    <a class="photo-box-inner fancybox" rel="fancybox-thumb" data-fancybox-group="gallery-undefined" title="" href="$2$3" target="_blank" style="top: -168.5px;">
                        <img $1 class="photo" src="$2$3" itemprop="image" $5>
                    </a>
                </div>', $content);





$content = preg_replace('#<img(.*?)src="([^"]*/)?(([^"/]*)\.[^"]*)"([^>]*?)>左(<br\s*/?>)?#',
'<div class="photo-box">
                    <a class="photo-box-inner fancybox" rel="fancybox-thumb" data-fancybox-group="gallery-undefined" title="" href="$2$3" target="_blank" style="top: -31px;">
                        <img class="photo" src="$2$3" itemprop="image"$5>
                    </a>
                </div>', $content);


$content = preg_replace('#<img(.*?)src="([^"]*/)?(([^"/]*)\.[^"]*)"([^>]*?)>右(<br\s*/?>)?#',
'<div class="photo-box">
                    <a class="photo-box-inner fancybox" rel="fancybox-thumb" data-fancybox-group="gallery-undefined" title="" href="$2$3" target="_blank" style="left: -18.9582px; top: -2.5px;">
                        <img class="photo" src="$2$3" itemprop="image"$5 style="height: 328px; width: 584.916px;">
                    </a>
                </div>', $content);
$content = preg_replace('#<img(.*?)src="([^"]*/)?(([^"/]*)\.[^"]*)"([^>]*?)>\.<br\s*/?>#',
'<div class="photo-box">
                    <a class="photo-box-inner fancybox" rel="fancybox-thumb" data-fancybox-group="gallery-undefined" title="" href="$2$3" target="_blank" style="top: -2px; left: -0.333333px;">
                        <img class="photo" src="$2$3" itemprop="image"$5 style="height: 328px; width: 546.667px;">
                    </a>
                </div>', $content);

$content = preg_replace('#<img(.*?)src="([^"]*/)?(([^"/]*)\.[^"]*)"([^>]*?)>\-(<br\s*/?>)?#',
'<div class="photo-box">
                    <a class="photo-box-inner fancybox" rel="fancybox-thumb" data-fancybox-group="gallery-undefined" title="" href="$2$3" target="_blank" style="top: -2px; left: -0.333333px;">
                        <img class="photo" src="$2$3" itemprop="image"$5 style="height: 328px; width: 546.667px;">
                    </a>
                </div></div></div>', $content);



$content = preg_replace('#{(.*?)\|(.*?)}#',
        '<ruby>$1<rp> (</rp><rt>$2</rt><rp>) </rp></ruby>', $content);
    return $content;
}
function get_post_view($archive)
{
    $cid    = $archive->cid;
    $db     = Typecho_Db::get();
    $prefix = $db->getPrefix();
    if (!array_key_exists('views', $db->fetchRow($db->select()->from('table.contents')))) {
        $db->query('ALTER TABLE `' . $prefix . 'contents` ADD `views` INT(10) DEFAULT 0;');
        echo 0;
        return;
    }
    $row = $db->fetchRow($db->select('views')->from('table.contents')->where('cid = ?', $cid));
    if ($archive->is('single')) {
 $views = Typecho_Cookie::get('extend_contents_views');
        if(empty($views)){
            $views = array();
        }else{
            $views = explode(',', $views);
        }
if(!in_array($cid,$views)){
       $db->query($db->update('table.contents')->rows(array('views' => (int) $row['views'] + 1))->where('cid = ?', $cid));
array_push($views, $cid);
            $views = implode(',', $views);
            Typecho_Cookie::set('extend_contents_views', $views); //记录查看cookie
        }
    }
    echo $row['views'];
}

function timer_start() {
global $timestart;
$mtime = explode( ' ', microtime() );
$timestart = $mtime[1] + $mtime[0];
return true;
}
timer_start();
 
function timer_stop( $display = 0, $precision = 3 ) {
global $timestart, $timeend;
$mtime = explode( ' ', microtime() );
$timeend = $mtime[1] + $mtime[0];
$timetotal = $timeend - $timestart;
$r = number_format( $timetotal, $precision );
if ( $display )
echo $r;
return $r;
}
/** 输出文章缩略图 */
function showThumbnail($widget)
{ 
    // 当文章无图片时的默认缩略图
   $dir = './usr/themes/Themia/img/sj/';
    $n=sizeof(scandir($dir))-2;
    $rand = rand(1,$n); 
    $random = $widget->widget('Widget_Options')->themeUrl . '/img/sj/' . $rand . '.jpg'; // 随机缩略图路径
   // $random = $widget->widget('Widget_Options')->themeUrl . '/img/mr.jpg'; // 若只想要一张默认缩略图请删除本行开头的"//"

    $attach = $widget->attachments(1)->attachment;
    $pattern = '/\<img.*?src\=\"(.*?)\"[^>]*>/i'; 
  $patternMD = '/\!\[.*?\]\((http(s)?:\/\/.*?(jpg|png))/i';
    $patternMDfoot = '/\[.*?\]:\s*(http(s)?:\/\/.*?(jpg|png))/i';

 if ($attach && $attach->isImage) {
      echo $attach->url; 
    } else
if (preg_match_all($pattern, $widget->content, $thumbUrl)) {
         echo $thumbUrl[1][0];
    }
//如果是内联式markdown格式的图片
    else if (preg_match_all($patternMD, $widget->content, $thumbUrl)) {
      echo $thumbUrl[1][0];
    }
    //如果是脚注式markdown格式的图片
    else if (preg_match_all($patternMDfoot, $widget->content, $thumbUrl)) {
      echo $thumbUrl[1][0];
    }
 else{
echo $random;
}

}

function theNext($widget, $default = NULL)
{
$db = Typecho_Db::get();
$sql = $db->select()->from('table.contents')
->where('table.contents.created > ?', $widget->created)
->where('table.contents.status = ?', 'publish')
->where('table.contents.type = ?', $widget->type)
->where('table.contents.password IS NULL')
->order('table.contents.created', Typecho_Db::SORT_ASC)
->limit(1);
$content = $db->fetchRow($sql);

if ($content) {
$content = $widget->filter($content);
$link = '<a class="post-action-btn btn btn--default tooltip--top" href="' . $content['permalink'] . '"  data-tooltip="' . $content['title'] . '">';
echo $link;
} else {
$link = '<a class="post-action-btn btn btn--disabled" target="_blank">
';
echo $link;
}
}
 
/**
* 显示上一篇
*
* @access public
* @param string $default 如果没有下一篇,显示的默认文字
* @return void
*/
function thePrev($widget, $default = NULL)
{
$db = Typecho_Db::get();
$sql = $db->select()->from('table.contents')
->where('table.contents.created < ?', $widget->created)
->where('table.contents.status = ?', 'publish')
->where('table.contents.type = ?', $widget->type)
->where('table.contents.password IS NULL')
->order('table.contents.created', Typecho_Db::SORT_DESC)
->limit(1);
$content = $db->fetchRow($sql);
 
if ($content) {
$content = $widget->filter($content);
$link = '<a class="post-action-btn btn btn--default tooltip--top" href="' . $content['permalink'] . '"  data-tooltip="' . $content['title'] . '">';
echo $link;
} else {
$link = '<a class="post-action-btn btn btn--disabled" target="_blank">
';
echo $link;
}
}
